components/ai_chat/resources/untrusted_conversation_frame/components/assistant_task/use_extract_task_data.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+import * as Mojom from '../../../common/mojom'
+
+// TODO(https://github.com/brave/brave-browser/issues/48535):
+// Should be Navigate and TODO tools.
+const importantToolNames = [Mojom.NAVIGATE_TOOL_NAME]
+
+export default function useExtractTaskData(
+  assistantEntryGroup: Mojom.ConversationTurn[],
+) {
+  return React.useMemo(() => {
+    // Individual tasks, split by CompletionEvent
+    let taskItems: Mojom.ConversationEntryEvent[][] = [[]]
+    // All completion events are allowed the links provided by the whole response
+    // group.
+    const allowedLinks: string[] = []
+    // Most recent of each type of important tool use event
+    const importantToolUseEvents: Partial<
+      Record<(typeof importantToolNames)[number], Mojom.ToolUseEvent>
+    > = {}
```

Comment:
Hey does this work? `importantToolNames` isn't declared as `as const` so I supect this is a record of `<string, Mojom.ToolUseEvent>`

Probably we want an `as const` on the declaration

Reply:
The `as const` causes issues when using it for `importantToolNames.includes(event.toolUseEvent.toolName)`. You're right in that it's just treating it as a string, so I could just be explicit about that. It's not necessary to type the key, just to know it is a valid tool name, which is a string.

Reply:
changed in 246a466